LOGGING_STANDARDS.md

```diff
@@ -1,58 +1,150 @@
+| `span_id`    | No             | `string`                   |                              | [Span ID from OpenCensus](https://opencensus.io/tracing/span/spanid/)
+| `severity`   | No             | `int8`                     | `2`                          | [The event severity level code](#severity-levels)
+| `http`       | No             | [`http`](#http-event-data) |                              | [HTTP event data](#http-event-data)
+| `auth`       | No             | [`auth`](#auth-event-data) |                              | [Authentication event data](#auth-event-data)
+| `data`       | No             | `object`<sup>3</sup>       |                              | Arbitrary key-value pairs (see below for a full explanation)
+
+<sup>1</sup> All dates must be UTC and in ISO8601 extended date time format: `yyyy-MM-dd'T'HH:mm:ss.SSSZZ` (e.g. `2019-01-21T16:19:12.356Z`)
+
+<sup>2</sup> Not mandatory, but must be included for all events created during the handling of a request.
+
+<sup>3</sup> The event-specific details in `data` are input as an object/map in code, but are stored as a text string by the centralised logging service to allow additional detail to be added to an event without needing to worry about key name collision.  These details can still be searched using a general text search on the `data` field.
+
+#### HTTP event data
+
+HTTP events are so common that we've defined a specific top-level field to capture it. The `http` field defines the common http event data.
+
+This can be used to log inbound or outbound HTTP event data.
+
+| Field name    | Required | Type                   | Example                      | Description
+| ------------- | -------- | ---------------------- | ---------------------------- | -----------
+| `method`      | Yes      | `string`               | `"GET"`                      | The HTTP method used
+| `path`        | Yes      | `string`               | `"/healthcheck"`             | The request path from the URL
+| `query`       | Yes      | `string`               | `"x=1&y=1"`                  | The unparsed query string from the URL
+| `scheme`      | Yes      | `string`               | `"https"`                    | The scheme or protocol from the URL
+| `host`        | Yes      | `string`               | `"10.100.20.1"`              | The hostname or ip from the URL
+| `port`        | Yes      | `int32`                | `443`                        | The port number from the URL
+| `status_code` | No       | `int16`                | `200`                        | The HTTP status code
+| `started_at`  | No       | `datetime`<sup>1</sup> | `"2019-01-21T16:19:12.356Z"` | ISO8601 date string of the request start time
+| `ended_at`    | No       | `datetime`<sup>1</sup> | `"2019-01-21T16:19:12.356Z"` | ISO8601 date string of the request end time
+| `duration`    | No       | `int64`                | `236578`                     | Difference between `started_at` and `ended_at` in nanoseconds
+
```

Comment:
We need to discuss error / exception information. Open to debate but for now I would suggest something like:
``` 
{
  ...
  "error": { 
    "cause": "something bad happened",
    "stack_trace": "STACK_TRACE_AS_STRING"
  }
  ...
}